frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts

```diff
@@ -385,28 +321,170 @@ export class MonitoredMetricsComponent implements OnInit, OnChanges, OnDestroy {
     return metric;
   }
 
-  checkQueryNameError(queryName){
+  checkMetricKeyRequiredError(metricKeyMissing: Boolean){
+    if (metricKeyMissing) {
+      this.queryMetricKeyError.push(true);
+    }
+  }
+
+  checkQueryNameRequiredError(queryName: any){
     if (queryName == null || queryName === '') {
       this.queryNameError.push(true);
     }
   }
 
+  checkStatisticRequiredError(statistic: any) {
+    if (statistic == null || statistic === '') {
+      this.queryStatisticError.push(true);
+    }
+  }
+
+  checkComparisonStatisticRequiredError(compareFn: any, compareValue: any) {
+    if (compareFn == null || compareFn === '' || compareValue == null || compareValue === '') {
```

Comment:
i'm seeing a number of places with null-checking specific things, but this is actually less type-safe than just using the existence-checking pattern of `if (!compareFn || !compareValue)`. Empty string `''` is a falsey value, just like `null`, they can both be covered this way. The current code will allow `undefined`, `NaN`, `0`, `false` values to slip through.